src/app/modules/item/task-communication/rxjschannel.ts

```diff
@@ -35,31 +40,70 @@ export class RxMessagingChannel {
     return this.innerChan.unbind(method, doNotPublish);
   }
 
-  bind<T>(method: string, observable?: (params: T) => Observable<unknown>, doNotPublish?: boolean): MessagingChannel {
+  /** Bind a local method, allowing the remote task to call it */
+  bind<T>(method: string, observable?: (params: T) => Observable<unknown>, validator?: D.Decoder<unknown, T>,
+    selector?: (params: any[]) => unknown, doNotPublish?: boolean): MessagingChannel {
     // Create a callback wrapping the observable bound
-    function callback(transaction: MessageTransaction, params: T): void {
+    function callback(transaction: MessageTransaction, ...params: any[]): void {
       if (!observable) {
         return;
       }
-      const cb$ = observable(params);
+      // Select params
+      const actualSelector = selector
+        ? selector
+        : (result: any[]) : unknown => (result.length > 0 ? result[0] : undefined);
+
+      const selectedParams = actualSelector(params);
+
+      // Validate params before passing them, if there is a validator
+      const decodedParams = validator
+        ? fppipe(
+          validator.decode(selectedParams),
+          fold(
+            error => {
+              throw new Error(D.draw(error));
+            },
+            decoded => decoded
+          ))
+        : selectedParams as T;
```

Comment:
Note to self:
Same as the selector logic, to me it should be done via rxjs, the method should not internalise decoders